js/widgets/navbar.js

```diff
@@ -35,59 +35,112 @@ return $.widget( "mobile.navbar", {
-		$navbar.addClass( "ui-navbar" )
+		navbar.addClass( "ui-navbar" )
 			.attr( "role", "navigation" )
 			.find( "ul" )
-				.jqmEnhanceable()
-				.grid( { grid: this.options.grid } );
-
-		$navbuttons
-			.each( function() {
-				var icon = $.mobile.getAttribute( this, "icon" ),
-					theme = $.mobile.getAttribute( this, "theme" ),
-					classes = "ui-button";
-
-				if ( theme ) {
-					classes += " ui-button-" + theme;
-				}
-				if ( icon ) {
-					classes += " ui-icon-" + icon + " ui-button-icon-" + iconpos;
-				}
-				$( this ).addClass( classes );
+			.jqmEnhanceable();
+
+		$.extend( self, {
+			navbar: navbar,
+			navButtons: navButtons,
+			numButtons: numButtons,
+			maxButton: maxButton,
+			iconpos: iconpos
+		} );
+
+		 if ( numButtons <= maxButton ) {
+			navButtons.each( function() {
+				self._makeNavButton( this, iconpos );
 			} );
+		} else {
+			self._createNavRows();
+		}
 
-		$navbar.delegate( "a", "vclick", function( /* event */ ) {
-			var activeBtn = $( this );
+	},
 
-			if ( !( activeBtn.hasClass( "ui-state-disabled" ) ||
+	_createNavRows: function() {
+		var rowCount, row, pos, buttonItem, overflowNav, navRow,
+			navItems = this.navbar.find( "li" ),
+			buttonCount = this.numButtons,
+			maxButton = this.maxButton;
+
+		rowCount = ( buttonCount % maxButton ) === 0 ?
+						( buttonCount / maxButton ) :
+						Math.floor( buttonCount / maxButton ) + 1;
+
+		// prep for new rows
+		for ( pos = 1; pos < rowCount ; pos++ ) {
+			navRow = $( "<ul>" );
+			this._addClass( navRow, "ui-navbar-row ui-navbar-row-" + pos );
+			navRow.appendTo( this.navbar );
+		}
+
+		// enhance buttons and move to new rows
+		for ( pos = 0; pos < buttonCount ; pos++ ) {
+			buttonItem = navItems.eq( pos );
+			this._makeNavButton( buttonItem.find( "a" ), this.iconpos );
+			if ( pos + 1 > maxButton ) {
+				buttonItem.detach();
+				row = ( ( pos + 1 ) % maxButton ) === 0 ?
+						Math.floor( ( pos ) / maxButton ) :
+						Math.floor( ( pos + 1 ) / maxButton );
+				overflowNav = "ul.ui-navbar-row-" + row;
+				this.navbar.find( overflowNav ).append( buttonItem );
+			}
+		}
+	},
 
-					// DEPRECATED as of 1.4.0 - remove after 1.4.0 release
-					// only ui-state-disabled should be present thereafter
-					activeBtn.hasClass( "ui-disabled" ) ||
-					activeBtn.hasClass( "ui-button-active" ) ) ) {
+	_makeNavButton: function( button, iconpos ) {
+		$( button ).button( { iconPosition: iconpos } );
+	},
 
-				$navbuttons.removeClass( "ui-button-active" );
-				activeBtn.addClass( "ui-button-active" );
+	refresh: function() {
+		var self = this;
 
-				// The code below is a workaround to fix #1181
-				$( document ).one( "pagehide", function() {
-					activeBtn.removeClass( "ui-button-active" );
-				} );
-			}
-		} );
+		self.navButtons = self.navbar.find( "a" );
+		self.numButtons = self.navButtons.length;
+
+		this._addClass( self.navbar, "ui-navbar" );
+		self.navbar.attr( "role", "navigation" )
+			.find( "ul" )
+			.jqmEnhanceable();
```

Comment:
`jqmEnhanceable` is deprecated and should this be `find` or should it be `children` do we support multiple lists?